### PR TITLE
introduce an otelcol.exporter.kafka component

### PR DESCRIPTION
**Pull request by tpaschalis**
#### PR Description
This component adds the otelcol.exporter.kafka component so that Grafana Agent can work with Kakfa pipelines end-to-end.

#### Which issue(s) this PR fixes
No issue files

#### PR Checklist

- [X] CHANGELOG.md updated
- [X] Documentation added
- [X] Tests updated
- [x] Config converters updated

### CODE REVIEW

File: internal/component/otelcol/exporter/kafka/kafka.go [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+	"github.com/open-telemetry/opentelemetry-collector-contrib/exporter/kafkaexporter"
+	otelcomponent "go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/exporter/exporterhelper"
+	otelextension "go.opentelemetry.io/collector/extension"
+)
+
+func init() {
+	component.Register(component.Registration{
+		Name:      "otelcol.exporter.kafka",
+		Stability: featuregate.StabilityStable,
```
Comment (rfratto) [inline internal/component/otelcol/exporter/kafka/kafka.go:24]: Is this the right stability?

File: docs/sources/flow/reference/components/otelcol.exporter.kafka.md [hunk 2]
```diff
@@ -0,0 +1,328 @@
...
+---
+
+# otelcol.exporter.kafka
+
+`otelcol.exporter.kafka` accepts telemetry data from other `otelcol` components
+and writes them over to a Kafka topic.
+
+> **NOTE**: `otelcol.exporter.kafka` is a wrapper over the upstream
+> OpenTelemetry Collector `kafka` exporter. Bug reports or feature requests will
+> be redirected to the upstream repository, if necessary.
```
Comment (clayton-cornell) [inline docs/sources/flow/reference/components/otelcol.exporter.kafka.md:14]:
```suggestion
{{< admonition type="note" >}}
`otelcol.exporter.kafka` is a wrapper over the upstream OpenTelemetry Collector `kafka` exporter.
Bug reports or feature requests will be redirected to the upstream repository, if necessary.
{{< /admonition >}}
```

File: docs/sources/flow/reference/components/otelcol.exporter.kafka.md [hunk 3]
```diff
@@ -0,0 +1,328 @@
...
+
+This component uses a synchronous producer that blocks and does not batch
+messages, therefore it should be used with batch and queued retry processors
```
Comment (clayton-cornell) [inline docs/sources/flow/reference/components/otelcol.exporter.kafka.md:17]:
```suggestion
messages. Therefore, it should be used with batch and queued retry processors
```

File: docs/sources/flow/reference/components/otelcol.exporter.kafka.md [hunk 4]
```diff
@@ -0,0 +1,328 @@
...
+prepended to it, for higher throughput and resiliency.
```
Comment (clayton-cornell) [inline docs/sources/flow/reference/components/otelcol.exporter.kafka.md:18]:
```suggestion
prepended to it for higher throughput and resiliency.
```

**Review by clayton-cornell — COMMENTED** [review]
Some minor doc suggestions

File: internal/component/otelcol/exporter/kafka/kafka.go [hunk 5]
```diff
@@ -0,0 +1,165 @@
...
+	Encoding string        `river:"encoding,attr,optional"`
+	Timeout  time.Duration `river:"timeout,attr,optional"`
+
+	Queue        otelcol.QueueArguments        `river:"sending_queue,block,optional"`
+	Retry        otelcol.RetryArguments        `river:"retry_on_failure,block,optional"`
+	DebugMetrics otelcol.DebugMetricsArguments `river:"debug_metrics,block,optional"`
+
+	Auth     kafka.AuthenticationArguments `river:"authentication,block,optional"`
+	Metadata kafka.MetadataArguments       `river:"metadata,block,optional"`
+	Producer Producer                      `river:"producer,block,optional"`
```
Comment (wildum) [inline internal/component/otelcol/exporter/kafka/kafka.go:51]: this block is missing in the documentation

File: docs/sources/flow/reference/components/otelcol.exporter.kafka.md [hunk 6]
```diff
@@ -0,0 +1,328 @@
...
+}
+```
+
+## Arguments
+
+`otelcol.exporter.kafka` supports the following arguments:
+
+Name | Type | Description | Default | Required
+---- | ---- | ----------- | ------- | --------
+`protocol_version` | `duration` | Kafka protocol version to use. | | yes
```
Comment (wildum) [inline docs/sources/flow/reference/components/otelcol.exporter.kafka.md:37]:
```suggestion
`protocol_version` | string | Kafka protocol version to use. | | yes
```

File: docs/sources/flow/reference/components/otelcol.exporter.kafka.md [hunk 7]
```diff
@@ -0,0 +1,328 @@
...
+`brokers` | list(string) | Kafka brokers to connect to. | `["localhost:9092"]` | no
+`client_id` | string | The client ID to configure the Sarama Kafka client with for all produce requests. | `"sarama"` | no
+`topic` | string | The name of the Kafka topic to export to. | see below | no
```
Comment (wildum) [inline docs/sources/flow/reference/components/otelcol.exporter.kafka.md:40]: To what is this "see below" referring?

File: docs/sources/flow/reference/components/otelcol.exporter.kafka.md [hunk 8]
```diff
@@ -0,0 +1,328 @@
...
+`encoding` | string | The encoding of data sent to Kafka. | see below | no
```
Comment (wildum) [inline docs/sources/flow/reference/components/otelcol.exporter.kafka.md:41]: I think that we should still say that otlp_proto is the default

**Comment by rfratto** [comment]
I'm going to close this; the otelcol.exporter.kafka component in Alloy has taken precedence.

**Pull request closed without merging**